agent/dockerclient/dockerapi/docker_client.go

```diff
@@ -1372,25 +1371,35 @@ func (dg *dockerGoClient) Stats(ctx context.Context, id string, inactivityTimeou
 		}()
 	} else {
 		seelog.Infof("DockerGoClient: Starting to Poll for metrics for container %s", id)
-		//Sleep for a random period time up to the polling interval. This will help make containers ask for stats at different times
-		time.Sleep(time.Second * time.Duration(rand.Intn(int(dg.config.PollingMetricsWaitDuration.Seconds()))))
-
+		// firstStat jitters the time at which containers ask for their first stat.
+		firstStat := time.After(retry.AddJitter(time.Nanosecond, dg.config.PollingMetricsWaitDuration/2))
+		// sleeping here jitters the time at which the ticker is created, so that
+		// containers do not synchronize on calling the docker stats api.
+		// the max sleep is 80% of the polling interval so that we have a chance to
+		// get two stats in the first publishing interval.
+		time.Sleep(retry.AddJitter(time.Nanosecond, dg.config.PollingMetricsWaitDuration*8/10))
 		statPollTicker := time.NewTicker(dg.config.PollingMetricsWaitDuration)
 		go func() {
 			defer cancelRequest()
 			defer close(statsC)
 			defer statPollTicker.Stop()
 
-			for range statPollTicker.C {
-				// ContainerStats outputs an io.ReadCloser and an OSType
+			for {
+				select {
+				case _, ok := <-statPollTicker.C:
+					if !ok {
+						return
+					}
+				case <-firstStat:
+				}
```

Comment:
I'm not fully clear on what this is doing -- are we tossing out the first stat?

Reply:
No firstStat is a channel returned from time.After, this select statement is waiting on either the polling ticker channel or the firstStat channel to fire.

Reply:
I can rename to firstStatC if that makes it more clear?

Reply:
ok, so we ignore firings from time.After? 
I'd add a comment like:
`// this select statement is waiting on either the polling ticker channel or the firstStat channel (from time.After) to fire`

Reply:
sure will do